rllib/algorithms/ppo/tf/ppo_tf_rl_module.py

```diff
@@ -45,13 +35,17 @@ def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
         if STATE_OUT in encoder_outs:
             output[STATE_OUT] = encoder_outs[STATE_OUT]
 
-        # Value head
+        # Value head.
+        # TODO (sven): Once we don't use Policy anymore for PPO (on new EnvRunner API),
```

Comment:
For this exact reason we should not merge compute vf in forward inference and we should keep that separate from forward explore. Why are we merging the two?

Reply:
B/c right now, the new stack is NOT usable for anyone configuring `explore=False` in their config. Which means it's currently breaking a lot of use cases, where users do greedy inference on a PPO module.

Agreed that this is only a temporary fix, but we need to quickly make this work.

Once PPO moves to the EnvRunner API, we will move vf computations completely outside of all forward passes and handle it where it should be handled (on the learner or algorithm end).

Reply:
I don't think the temporary fix should be here. There should be a place outside of this where conditioned on `explore=True` or `explore=False` it will call `forward_exploration` or `forward_inference`. This is a hacky fix which goes against the design principles of RLModules.

Reply:
So why did this violation against the design principles of RLModule make it into the library in the first place?

In other words: Why were `forward_exploration()` and `forward_inference()` already (before this PR) behaving so differently and return either `vf_preds` or not returning `vf_preds`? How would an unsuspecting user have known that only `forward_exploration()` returns `vf_preds`, but - for some magic reason - `forward_inference()` does not?
Bottom line: The root cause of this problem is being addressed right now by one of Simon's PR's, in which we make PPO independent of any old Policy class and can then perform the vf computations completely independently of the `forward_inference/exploration` passes.

Reply:
`forward_inference` is for running inference (when the model training is done) so it doesn't need the vf; you just need actions. `forward_exploration` is the forward method you use during sampling but during training. That's the mental model. 

Regarding the bottom-line, shouldn't we wait for that PR to land and fix the root cause instead of this fix?

Reply:
I disagree about the mental model, though. Both `forward_inference()` and `forward_exploration()` should be usable interchangeably within a sampling procedure supposed to collect data for training.

This is besides your point about fixing the root cause, which I 100% agree with, but it's a larger process that may take another couple of days/week(s) to be fully functional and tested. So in the meantime, can we just merge this fix and allow PPO to run normally again? It's currently broken for any user that wants to set `explore=False` in the eval workers, which is a completely sensible thing to do! This is not a rare usecase and therefore I made this issue P1:
https://github.com/ray-project/ray/issues/39791